phoenix-core/src/main/java/org/apache/phoenix/coprocessor/UngroupedAggregateRegionObserver.java

```diff
@@ -1111,7 +1125,7 @@ private RegionScanner rebuildIndices(final RegionScanner innerScanner, final Reg
             synchronized (innerScanner) {
                 do {
                     List<Cell> results = new ArrayList<Cell>();
-                    hasMore = innerScanner.nextRaw(results);
+                    hasMore =  newScanner.nextRaw(results);
```

Comment:
is anything else later in the pipeline assuming that innerScanner will get read fully here?

Reply:
Yes, the above changes will be no-op for partial rebuilds by MetaDataRegionObserver and GlobalIndexChecker. The changes are only for rebuilds by IndexTool and index create (both using ServerIndexBuildComplier).